src/shared/fileentry.cpp

```diff
@@ -185,15 +185,15 @@ bool FileEntry::isFromArchive(std::wstring archiveName) const
   std::scoped_lock lock(m_OriginsMutex);
 
   if (archiveName.length() == 0) {
-    return m_Archive.first.length() != 0;
+    return m_Archive.isValid();
   }
 
-  if (m_Archive.first.compare(archiveName) == 0) {
+  if (m_Archive.name().compare(archiveName) == 0) {
     return true;
   }
 
   for (auto alternative : m_Alternatives) {
-    if (alternative.second.first.compare(archiveName) == 0) {
+    if (alternative.archive().name().compare(archiveName) == 0) {
```

Comment:
Same here (case-insensitive)?

Reply:
maybe the values are assumed already lowercase? I'll check the code

Reply:
I don't think case is ever checked anywhere for archive names, I'd leave it as-is for now, at least for this PR.